src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/ForwardCancellationTokenToInvocations.Fixer.cs

```diff
@@ -0,0 +1,199 @@
+                    paramName = argument.Parameter.Name;
+                }
+            }
+
+            // Create and append new ct argument to pass to the invocation, using the ancestor method parameter name
+            SyntaxNode cancellationTokenIdentifier = generator.IdentifierName(cancellationTokenParameterName);
+            SyntaxNode cancellationTokenNode;
+            if (shouldTokenUseName)
+            {
+                // If the paramName is unknown at this point, it's because an overload contains the ct parameter
+                // and since it cannot be obtained, no fix will be provided or else CA8323 shows up:
+                // CA8323: Named argument 'argName' is used out-of-position but is followed by an unnamed argument
+                if (string.IsNullOrEmpty(paramName))
+                {
+                    return false;
+                }
+
+                cancellationTokenNode = generator.Argument(paramName, RefKind.None, cancellationTokenIdentifier);
+            }
+            else
+            {
+                cancellationTokenNode = generator.Argument(cancellationTokenIdentifier);
+            }
+            newArguments.Add(cancellationTokenNode);
+
+            SyntaxNode newInvocation;
+            // The instance is null when calling a static method from another type
+            if (invocation.Instance == null)
+            {
+                SyntaxNode staticType = generator.TypeExpressionForStaticMemberAccess(invocation.TargetMethod.ContainingType);
+                newInvocation = generator.MemberAccessExpression(staticType, invocation.TargetMethod.Name);
+            }
+            // The method is being invoked with nullability
+            else if (invocation.Instance is IConditionalAccessInstanceOperation)
+            {
+                newInvocation = GetConditionalOperationInvocationExpression(invocation.Syntax);
+            }
+            // The instance is implicit when calling a method from the same type, call the method directly
+            else if (invocation.Instance.IsImplicit)
+            {
+                newInvocation = invocation.GetInstanceSyntax()!;
+            }
+            // Calling a method from an object, we must include the instance variable name
+            else
+            {
+                newInvocation = generator.MemberAccessExpression(invocation.GetInstanceSyntax(), invocation.TargetMethod.Name);
+            }
+            // Insert the new arguments to the new invocation
+            SyntaxNode newInvocationWithArguments = generator.InvocationExpression(newInvocation, newArguments).WithTriviaFrom(invocation.Syntax);
+
+            newRoot = generator.ReplaceNode(root, invocation.Syntax, newInvocationWithArguments);
+
+            return true;
+        }
+
+        // Needed for Telemetry (https://github.com/dotnet/roslyn-analyzers/issues/192) 
+        private class MyCodeAction : DocumentChangeAction
```

Comment:
@mavasani I am not sure to understand when this needs to be used and when it doesn't. Could you help?

Reply:
This is required for telemetry for this code fix being triggered.